File: data/2/OOP/Gyak1/Gyak1/Point.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Gyak1
{
    internal class Point
    {
        private readonly double x;
        private readonly double y;

        public Point(double a, double b) { x = a; y = b; }

        public static double Distance(Point p, Point q)
        {
            return Math.Sqrt(Math.Pow(p.x - q.x, 2) + Math.Pow(p.y - q.y, 2));      
        }
    }
}

File: data/2/ProgNyelvek/Gyak02/Point.java
public class Point
{
    double x;
    double y;

    public void move(double dx, double dy)
    {
        this.x += dx;
        this.y += dy;
    }

    Point(double a, double b)
    {
        this.x = a;
        this.y = b;
    }

    public void mirror(Point p)
    {
        this.x += 2*(p.x - this.x);
        this.y += 2*(p.y - this.y);  
    }

    public double distance(Point p)
    {
        return Math.sqrt(((p.x - this.x) * (p.x - this.x)) + ((p.y - this.y) * (p.y - this.y)));
    }
}